homeassistant/components/climate/flexit.py

```diff
@@ -0,0 +1,274 @@
+"""
+Platform for Flexit AC units with CI66 Modbus adapter.
+
+Example configuration:
+
+climate:
+  - platform: flexit
+    name: Main AC
+    slave: 21
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/climate.flexit/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_NAME, CONF_SLAVE, TEMP_CELSIUS,
+    ATTR_TEMPERATURE, DEVICE_DEFAULT_NAME)
+from homeassistant.components.climate import (ClimateDevice, PLATFORM_SCHEMA)
+import homeassistant.components.modbus as modbus
+import homeassistant.helpers.config_validation as cv
+
+DEPENDENCIES = ['modbus']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SLAVE): vol.All(int, vol.Range(min=0, max=32)),
+    vol.Optional(CONF_NAME, default=DEVICE_DEFAULT_NAME): cv.string
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+REGMAP_INPUT = {
```

Comment:
This is too much protocol specific information. Please extract it into a third party lib.